doc/arch/adr-002-decide-how-to-display-more-facilities.md

```diff
@@ -0,0 +1,232 @@
+limited experience using Rust. Together this means a Martin-based tile server
+may be difficult to operate and debug.
+
+### Adding a Vector Tile Endpoint to the Existing Django Web App
+
+Adding a vector tile endpoint to the existing web app seemed like a promising
+approach, since it would enable trying out `ST_AsMVT` while reusing the app's
+database connection and Django's querysets for filtering. In this approach we
+would add a `/tile/{layer}/{z}/{x}/{y}/` endpoint to the Django application,
+then update the client to make tile requests there rather than rendering the
+`/facilities` GeoJSON response as Leaflet markers.
+
+#### Pros
+
+##### Provides Access to the Existing Django Queryset Apparatus
+
+While using Martin (or a similar solution) would compel writing new versions of
+the facilities queries in PL/pgSQL, placing a vector tile endpoint in Django
+lets us reuse some of the existing query code and also provides access to
+Django models and querysets. Likewise, we would not have to write new code for
+new filter and search options in two different languages.
+
+##### Already Has a Secure Database Connection
+
+The Django web application already has a secure database connection, so we
+would not have to create a solution for securing Martin or another PostGIS-backed
+tile server.
+
+##### Enables Switching from `ST_AsMVT` to Another Python Vector Tile Option
+
+There remains some question about the viability of using `ST_AsMVT`. If it turns
+out that this is not a performant solution, having the tile endpoint in Django
+makes it possible to drop out of using `ST_AsMVT` altogether and instead using
+an alternate Python library for generating the vector tiles.
+
+##### Doesn't Require Creating & Deploying a Different Service
+
+Adding Martin or another vector tile server would increase the number of
+different kinds of services running as part of the OAR, which adds to the
+application's complexity. Keeping the tile endpoint in Django does not require
+adding a new service.
+
+##### Allows Scaling by Increasing the Number of App Instances
+
+Adding a tile endpoint to the Django app also enables continuing to scale the
+application in the usual way: by increasing the number of app instances
+available to serve requests.
+
+#### Cons
+
+##### Mingles Tile Request Traffic with Other App Traffic
+
+The biggest downside of adding a vector tile endpoint to the Django app is
+that it would mean mingling tile request traffic with other app traffic.
+While we plan to limit the size of tile request responses by clusting facilities
+at different zoom levels, tile request traffic will likely be more frequent and
+sustained than requests to the current `/facilities` endpoint.
+
+## Decision
+
+We have decided to add a vector tile endpoint to the existing Django app.
+
+While Martin, in particular, seemed like a compelling solution, we had enough
+questions about using it to discourage us from taking on the complexity of
+using it here.
```

Comment:
I am very curious about Martin (et al), and I think others are too! We discussed that this would be a good R&D opportunity. Can those who are more involved with this project explain why this project in and of itself can't be that R&D opportunity? What are the limitations I don't know about, and can they be further discussed/captured?

Reply:
@jwalgran may amplify or correct this but I think one main reason not to have it here is that the client needs to be able to request subsets of facilities using query parameters like contributor id or facility name etc.

If we added Martin, making changes to these queries (like adding an ability to search on facility description or certifications) would require us to update both the Django queryset filter code and also the PL/pgSQL code used in Martin.

Reply:
Yep. If we didn't have to filter, Martin would be a more attractive choice. Writing our own endpoint also allows us to bail out of using `ST_AsMVT` if it proves to be a problem and do our tile rendering on an app server rather than in the database.

Reply:
Thanks, makes sense. Consider adding the above to the ADR